alacritty/src/display/content.rs

```diff
@@ -48,7 +48,7 @@ impl<'a> RenderableContent<'a> {
 
         // Copy the cursor and override its shape if necessary.
         let mut terminal_cursor = terminal_content.cursor;
-        if !show_cursor {
+        if !show_cursor || terminal_cursor.shape == CursorShape::Hidden {
```

Comment:
It's generally safe to check like that, since the old logic that checks for Vi mode is in a different place, though, I'm not sure whether this is maintainable, since haven't familiar much with refactored codebase.

Reply:
Should be fine. It's a little confusing that `show_cursor` doesn't include this but since that's coming from the UI (it's used for blinking), that should be fine.